modin/numpy/arr.py

```diff
@@ -408,71 +409,88 @@ def max(
         if self._ndim == 1:
             if axis == 1:
                 raise numpy.AxisError(1, 1)
-            result = self._query_compiler.max(axis=0)
+            if not isinstance(where, array):
+                result = self._query_compiler.max(axis=0)
+            else:
+                result = where.where(self, initial)._query_compiler.max(axis=0)
```

Comment:
I think this is an important case, we need to test it.

Reply:
Makes sense to me!